gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDatasetTest.java

```diff
@@ -77,80 +74,112 @@ public void testGetFilePaths() throws IOException {
+    mapOfPathAndFileStatus.put(path1, fileStatus1);
+    mapOfPathAndFileStatus.put(path2, fileStatus2);
+    mapOfPathAndFileStatus.put(path3, fileStatus3);
+    mapOfPathAndFileStatus.put(path4, fileStatus4);
+    mapOfPathAndFileStatus.put(path5, fileStatus5);
 
-    Mockito.when(fs.makeQualified(any(Path.class))).thenReturn(new Path("/root/iceberg/test/destination/sub_path_destination"));
-    Mockito.when(fs.getFileStatus(path1)).thenReturn(fileStatus1);
-    Mockito.when(fs.getFileStatus(path2)).thenReturn(fileStatus2);
-    Mockito.when(fs.getFileStatus(path3)).thenReturn(fileStatus3);
-    Mockito.when(fs.getFileStatus(path4)).thenReturn(fileStatus4);
-    Mockito.when(fs.getUri()).thenReturn(new URI(null, null, "/root/iceberg/test/output", null));
+    mockFileSystemMethodCalls(fs, mapOfPathAndFileStatus, test_qualified_path, test_uri_path);
 
     Collection<CopyEntity> copyEntities = icebergDataset.generateCopyEntitiesForTableFileSet(copyConfiguration);
-    Assert.assertEquals(copyEntities.size(), setOfFilePaths.size());
+    verifyCopyEntities(copyEntities, expected);
+
+  }
+
+  private void verifyCopyEntities(Collection<CopyEntity> copyEntities, List<String> expected) {
+    List<String> actual = new ArrayList<>();
     for (CopyEntity copyEntity : copyEntities) {
       String json = copyEntity.toString();
       JsonObject jsonObject = new Gson().fromJson(json, JsonObject.class);
       JsonObject objectData =
           jsonObject.getAsJsonObject("object-data").getAsJsonObject("origin").getAsJsonObject("object-data");
       JsonObject pathObject = objectData.getAsJsonObject("path").getAsJsonObject("object-data").getAsJsonObject("uri");
       String filepath = pathObject.getAsJsonPrimitive("object-data").getAsString();
-      Assert.assertTrue(setOfFilePaths.contains(filepath));
-      setOfFilePaths.remove(filepath);
+      actual.add(filepath);
+    }
+    Assert.assertEquals(actual.size(), expected.size());
+    Assert.assertEqualsNoOrder(actual.toArray(), expected.toArray());
+  }
+
+  private void mockFileSystemMethodCalls(FileSystem fs, Map<Path, FileStatus> pathToFileStatus, String qualifiedPath, String uriPath)
+      throws URISyntaxException, IOException {
+
+    Mockito.when(fs.getUri()).thenReturn(new URI(null, null, uriPath, null));
+    for (Map.Entry<Path, FileStatus> entry : pathToFileStatus.entrySet()) {
+      Path path = entry.getKey();
+      FileStatus fileStatus = entry.getValue();
+      Mockito.when(fs.getFileStatus(path)).thenReturn(fileStatus);
+      Mockito.when(fs.makeQualified(path)).thenReturn(new Path(qualifiedPath));
     }
-    Assert.assertTrue(setOfFilePaths.isEmpty());
   }
 
   private static class MockedIcebergTable extends IcebergTable {
 
-    public MockedIcebergTable(TableOperations tableOps) {
-      super(tableOps);
+    String metadataPath;
+    String manifestListPath;
+    String manifestFilePath;
+    List<String> manifestListFilePaths;
```

Comment:
this mocking only works for a single manifest file (e.g. when the manifest list has only one entry).  let's generalize to the full model, which is a list of manifest files, each with their own list of listed (data) file paths.